src/cluster_legacy.c

```diff
@@ -2990,7 +2991,51 @@ int clusterProcessPacket(clusterLink *link) {
 
                 if (clusterNodeIsMaster(sender)) {
                     /* Master turned into a slave! Reconfigure the node. */
-                    clusterDelNodeSlots(sender);
+                    if (master && !memcmp(master->shard_id, sender->shard_id, CLUSTER_NAMELEN)) {
+                        /* 'sender' was a primary and was in the same shard as `master`, its new primary */
+                        if (sender->configEpoch > senderConfigEpoch) {
+                            serverLog(LL_WARNING,
+                                    "Ignore stale gossip message from %.40s (%s) in shard %.40s;"
```

Comment:
```suggestion
                                    "Ignore stale message from %.40s (%s) in shard %.40s;"
```
This isn't gossip, it's just a message right?

Reply:
I think it is a gossip message but sure either option works for me